### PR TITLE
Add validator for map coordinates

### LINKED ISSUE #663: Add coordinate validator for map data
Similar to #662 but related to `map` data (scanned map images) instead.

We have lots of problems with scanned map data that has coordinates that are in invalid formats; see for example:
- bb847xp3575
- bq428sj1172
- by513wd0839

@justinlittman also [made a report](https://github.com/sul-dlss/purl/issues/708#issuecomment-1782904904) that shows the breadth of coordinate data formats in scanned maps. 

This causes problems when the maps are released, but fail to index, and there's no way to know that except that they don't show up in Earthworks.

### PR DESCRIPTION
**Pull request by thatbudakguy**
Closes #663

### CODE REVIEW

File: spec/cocina/models/validators/map_coordinates_validator_spec.rb [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+  end
+
+  context 'when coordinates contains non-coordinate data' do
+    let(:props) do
+      {
+        externalIdentifier: 'druid:xq000jd3530',
+        description: {
+          subject: [
+            {
+              value: 'W 62°51ʹ00ʺ--just kidding!!',
```
Comment (lwrubel) [inline spec/cocina/models/validators/map_coordinates_validator_spec.rb:124]: For clarity about why it raises, maybe use a valid coordinate string along with the extra non-coordinate data?

**Comment by lwrubel** [comment]
To be executed, the `MapCoordinatesValidator` needs to be added to the list of validators here: https://github.com/sul-dlss/cocina-models/blob/b642e33073aeb0ce87d05a6d447730fc5c8f29ae/lib/cocina/models/validators/validator.rb#L17

**Review by lwrubel — CHANGES_REQUESTED** [review]
Once added to the list of validators, LGTM.

File: lib/cocina/models/validators/map_coordinates_validator.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        # Values of all non-empty 'subject' elements with type 'map coordinates'
+        def coordinates_values
+          @coordinates_values ||= Array(attributes.dig(:description, :subject))
+                                  .select { |subject| subject[:type] == 'map coordinates' }
+                                  .map { |subject| subject[:value] }
+        end
+
+        def validate_coordinates(coordinates)
+          raise ValidationError, "Missing map coordinates for #{druid}" if coordinates.blank?
+          raise ValidationError, "Invalid map coordinates for #{druid}: #{coordinates}" unless Stanford::Mods::Coordinate.new(coordinates).valid?
```
[the remark this answers is not in the document]
Comment (lwrubel) [reply lib/cocina/models/validators/map_coordinates_validator.rb:41]: I felt bad I didn't know this gem existed, until I realized you just created it. 😅 Can you say more about how it's useful to have a separate gem for the coordinate class alone? Makes it easier to use in various applications? Would we use it EarthWorks, etc too?

[the remark this answers is not in the document]
Comment (jcoyne) [reply lib/cocina/models/validators/map_coordinates_validator.rb:41]: We don't want to pull in all of the mods gem when we only want to use Cocina.  This allows us just to use the part we care about.

[the remark this answers is not in the document]
Comment (thatbudakguy) [reply lib/cocina/models/validators/map_coordinates_validator.rb:41]: I was just thinking it'd be nice not to have to pull in all the mods stuff. Thanks for doing this.

**Pull request closed without merging**